BUGFIX: Backend login plays nicely with other authentications

kitsunet opened the pull request:
This slightly modifies the backend login and redirections so that:

The `LoginController` at `/neos/login` now specifically checks if
access to the Neos backend is currently granted and not only if any
account is authenticated before trying to redirect to the backend.

The `BackendController` at `/neos` is no longer protected by
Policy but checks internally for access to the Neos backend.
Before with policy protection in case you had no active account you
would be redirected to the login due to the provider entry point, but
if you were authenticated and couldn't access the `indexAction` or
weren't authenticated with a Neos user you would end up
with a 403 error, while now you will get redirected to the login.

These behavior changes make it much easier to have the Neos backend
login co-exist with another login and be authenticated to both.

kitsunet committed:
BUGFIX: Backend login plays nicely with other authentications

This slightly modifies the backend login and redirections so that:

The ``LoginController`` at ``/neos/login`` now specifically checks if
access to the neos backend is currently granted and not only if any
account is authenticated before trying to redirect to the backend.

The ``BackendController`` at ``/neos`` is no longer protected by
Policy but checks internally for access to the neos backend.
Before with policy protection in case you had no active account you
would be redirected to the login due to the provider entry point, but
if you were authenticated and couldn't access the ``indexAction`` or
weren't authenticated with a Neos user you would end up
with a 403 error.

These behavior changes make it much easier to have the neos bakend login
co-exist with another login and be authenticated to both.

# Conflicts:
#	Neos.Neos/Classes/Controller/Backend/BackendController.php

johannessteu reviewed (approved):
by reading

dimaip commented:
Looks good by reading. I wonder what's up with the tests though.

rolandschuetz commented:
@kitsunet Thanks for the authentification fix. Please also adopt the test cases, currently  Neos\Neos\Tests\Functional\Controller\Backend\BackendControllerSecurityTest::indexActionIsDeniedForEverybody fails.

kitsunet committed:
Adjust tests for new redirect behavior

# Conflicts:
#	Neos.Neos/Classes/Controller/Backend/BackendController.php
#	Neos.Neos/Tests/Functional/Controller/Backend/BackendControllerSecurityTest.php

kitsunet commented:
I have the feeling I uncovered a problem with our tests. I will have a look with @kdambekalns . A short look indicates to me that in the browser request the authenticated role is no longer authenticated :/

kitsunet commented:
Note: Karsten and me found two problems related to functional testing that I'll fix, after that we can adjust the tests here to the (then correct) behavior and it should be fine...

kitsunet commented:
https://github.com/neos/neos-development-collection/pull/2391

kitsunet commented:
https://github.com/neos/flow-development-collection/pull/1516

daniellienert commented:
@kitsunet: Would you rebase this PR?

daniellienert commented:
Reopened to get feedback from travis.

albe commented:
Should we get this merged? Would probably need a rebase on 4.3 by now

kdambekalns commented:
Rebased and ressolved a conflict, let's see what happens… 🤞

kdambekalns commented:
Behat passes, unit tests fail – any immediate ideas, @kitsunet?

markusguenther commented:
I have rebased the change, and now we are on track of Neos 5.3.
Let's see what happens :)

mhsdesign commented:
Hmm a little to much time passed. This is not super easily rebaseable on say 9.0. Is this fix still necessary?
